src/main/java/be/seeseemelk/mockbukkit/entity/ChestBoatMock.java

```diff
@@ -0,0 +1,182 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import be.seeseemelk.mockbukkit.inventory.ChestInventoryMock;
+import org.bukkit.entity.ChestBoat;
+import org.bukkit.entity.Entity;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.loot.LootTable;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class ChestBoatMock extends BoatMock implements ChestBoat
+{
+
+	private boolean refillEnabled;
+	private long lastFilled = -1;
+	private Map<UUID, Long> lootedPlayers;
+	private long nextRefill = -1;
+
+	private final Inventory inventory;
+
+
+	/**
+	 * Constructs a new {@link ChestBoatMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public ChestBoatMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+		this.inventory = new ChestInventoryMock(this, 3 * 9);
+	}
+
+	@Override
+	public boolean isRefillEnabled()
+	{
+		return this.refillEnabled;
+	}
+
+	public void setRefillEnabled(boolean refillEnabled)
+	{
+		this.refillEnabled = refillEnabled;
+	}
+
+	@Override
+	public boolean hasBeenFilled()
+	{
+		return this.lastFilled != -1;
+	}
+
+	@Override
+	public boolean canPlayerLoot(@NotNull UUID player)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public boolean hasPlayerLooted(@NotNull UUID player)
+	{
+		return this.lootedPlayers != null && this.lootedPlayers.containsKey(player);
+	}
+
+	@Override
+	public @Nullable Long getLastLooted(@NotNull UUID player)
+	{
+		return this.lootedPlayers != null ? this.lootedPlayers.get(player) : null;
+	}
+
+	@Override
+	public boolean setHasPlayerLooted(@NotNull UUID player, boolean looted)
+	{
+		final boolean hasLooted = hasPlayerLooted(player);
+
+		if (this.lootedPlayers == null)
+		{
+			this.lootedPlayers = new HashMap<>();
+		}
+
+		if (hasLooted != looted)
+		{
+			if (looted)
+			{
+				this.lootedPlayers.computeIfAbsent(player, p -> System.currentTimeMillis());
+			}
+			else
+			{
+				this.lootedPlayers.remove(player);
+			}
+		}
+		return hasLooted;
+	}
+
+	@Override
+	public boolean hasPendingRefill()
+	{
+		return this.nextRefill != -1;
+	}
+
+	@Override
+	public long getLastFilled()
+	{
+		return this.lastFilled;
+	}
+
+	@Override
+	public long getNextRefill()
+	{
+		return this.nextRefill;
+	}
+
+	@Override
+	public long setNextRefill(long refillAt)
+	{
+		final long oldRefill = this.nextRefill;
+		this.nextRefill = refillAt;
```

Comment:
```suggestion
		this.nextRefill = (refillAt < -1) ? -1 : refillAt;
```
https://github.com/MockBukkit/MockBukkit/pull/866#discussion_r1345461088